plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/dlf/AlibabaDlfMetaStoreConfig.java

```diff
@@ -0,0 +1,629 @@
+
+import com.aliyun.datalake.metastore.common.CacheDataLakeMetaStoreConfig;
+import com.aliyun.datalake.metastore.common.ProxyMode;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+import java.util.Optional;
+
+public class AlibabaDlfMetaStoreConfig
+{
+    public Optional<String> hiveMetastoreAuthenticationType = Optional.empty();
+    public Optional<String> hiveMetastoreServiceClientPrincipal = Optional.empty();
+    public Optional<String> hiveMetastoreServicePrincipal = Optional.empty();
+    public Optional<String> hiveMetastoreClientKeytab = Optional.empty();
+    public Optional<Boolean> enableBitVector = Optional.empty();
+    public Optional<Integer> batchSizeForGetPartititon = Optional.empty();
+    private Optional<String> regionId = Optional.empty();
+    private Optional<String> endpoint = Optional.empty();
+    private Optional<String> logStore = Optional.empty();
+    private Optional<String> userId = Optional.empty();
+    private Optional<String> accessKeyId = Optional.empty();
+    private Optional<String> accessKeySecret = Optional.empty();
+    private Optional<String> defaultWarehouseDir = Optional.empty();
+    private Optional<AlibabaDlfConfig.AKMode> akMode = Optional.empty();
+    private boolean isNewStsMode = true;
+    private Optional<ProxyMode> proxyMode = Optional.empty();
+    private short pageSize = AlibabaDlfConfig.DEFAULT_CATALOG_PAGE_SIZE;
+    private int batchSize = AlibabaDlfConfig.DEFAULT_CATALOG_ACCURATE_BATCH_SIZE;
+    private int readTimeout = AlibabaDlfConfig.DEFAULT_CATALOG_SERVER_READ_TIMEOUT_MILLS;
+    private int connTimeout = AlibabaDlfConfig.DEFAULT_CATALOG_SERVER_CONN_TIMEOUT_MILLS;
+    private boolean deleteDirWhenDropSchema = AlibabaDlfConfig.DELETE_DIR_WHEN_DROP_SCHEMA_VALUE;
+    private Optional<Boolean> dbCacheEnable = Optional.empty();
+    private Optional<Integer> dbCacheSize = Optional.empty();
+    private Optional<Integer> dbCacheTTLMins = Optional.empty();
+    private Optional<Boolean> tbCacheEnable = Optional.empty();
+    private Optional<Integer> tbCacheSize = Optional.empty();
+    private Optional<Integer> tbCacheTTLMins = Optional.empty();
+    private Optional<Boolean> assumeCanonicalPartitionKeys = Optional.empty();
+
+    private Optional<Boolean> enableFileOperation = Optional.empty();
+    private Optional<Double> enableFileOperationGrayRate = Optional.empty();
+    private Optional<Boolean> enableRecordActionLog = Optional.empty();
+    private Optional<Boolean> enableRecordLog = Optional.empty();
+    private Optional<Integer> tableColStatsBatchSize = Optional.empty();
+
+    public Optional<String> getRegionId()
+    {
+        return regionId;
+    }
+
+    @Config(AlibabaDlfConfig.CATALOG_REGION_ID)
+    @ConfigDescription("Region Id for Data Lake Formation Catalog")
+    public AlibabaDlfMetaStoreConfig setRegionId(String regionId)
+    {
+        if (regionId == null || regionId.isEmpty()) {
```

Comment:
The normal way we do config is to set the field to the default value:
```java
private String regionId = "cn-hangzhou";
```
Then the setter can simply assign the value:
```java
this.regionId = regionId;
```
You can add standard [Bean Validation](https://beanvalidation.org/) annotations such as `@NotNull` or `@NotEmpty` on the getter. The config system will use the Bean Validator to ensure the object is valid after setting all the config values.